rust/src/wallet.rs

```diff
@@ -235,6 +235,18 @@ pub async fn open_channel(peer_info: PeerInfo, channel_amount_sat: u64) -> Resul
     .await
 }
 
+pub async fn force_close_channel() -> Result<()> {
+    let channel_manager = {
+        let lightning = &get_wallet()?.lightning;
+
+        lightning.channel_manager.clone()
+    };
```

Comment:
🙃 I guess this can be a one-liner?
```suggestion
    let channel_manager =  get_wallet()?.lightning.channel_manager.clone();
```

Reply:
Yep! I copy-pasted it from somewhere where dropping the `MutexGuard` was important. But it isn't the case here.

Reply:
Ah, NVM. This is the problem:

```
    Checking ten_ten_one v0.1.0 (/home/luckysori/github/itchysats/10101/rust)
warning: this `MutexGuard` is held across an `await` point
   --> rust/src/wallet.rs:310:28
    |
310 |     let channel_manager = &get_wallet()?.lightning.channel_manager;
    |                            ^^^^^^^^^^^^^
    |
    = note: `#[warn(clippy::await_holding_lock)]` on by default
    = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
   --> rust/src/wallet.rs:310:5
    |
310 | /     let channel_manager = &get_wallet()?.lightning.channel_manager;
311 | |
312 | |     lightning::force_close_channel(channel_manager.clone(), remote_node_id).await?;
313 | |
314 | |     Ok(())
315 | | }
    | |_^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock
```

Reply:
hmm... I think my suggestion had a  `.clone()`, which should not trigger the warning

Reply:
Still doesn't work with `.clone()` on a separate line.

The only alternative fix that I have found is:

```rust
let channel_manager = std::sync::Arc::clone(&get_wallet()?.lightning.channel_manager);
lightning::force_close_channel(channel_manager, remote_node_id).await?;
```

I think this is because it is equivalent to:

```rust
let channel_manager = std::sync::Arc::clone({ &get_wallet()?.lightning.channel_manager });
lightning::force_close_channel(channel_manager, remote_node_id).await?;
```

Or maybe the whole expression is moved into the function?

Either way the key is to not hold the `MutexGuard` across the await point.